src/Plugin/Styleguide/searchStyleguide.php

```diff
@@ -0,0 +1,272 @@
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Render\RendererInterface;
+use Drupal\search\SearchPluginManager;
+
+/**
+ * Search Styleguide items implementation.
+ *
+ * @Plugin(
+ *   id = "search_styleguide",
+ *   label = @Translation("Search Styleguide elements")
+ * )
+ */
+class SearchStyleguide extends StyleguidePluginBase {
+
+  /**
+   * The styleguide generator service.
+   *
+   * @var \Drupal\styleguide\Generator
+   */
+  protected $generator;
+
+  /**
+   * The theme manager service.
+   *
+   * @var \Drupal\Core\Theme\ThemeManagerInterface
+   */
+  protected $themeManager;
+
+  /**
+   * The module handler service.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * The form builder.
+   *
+   * @var \Drupal\Core\Form\FormBuilder
+   */
+  protected $formBuilder;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * An entity manager object.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * The Renderer service to format the username and node.
+   *
+   * @var \Drupal\Core\Render\RendererInterface
+   */
+  protected $renderer;
+
+  /**
+   * The search manager.
+   *
+   * @var \Drupal\search\SearchPluginManager
+   */
+  protected $searchManager;
+
+  /**
+   * Constructs a new searchStyleguide.
+   *
+   * @param array $configuration
+   * @param string $plugin_id
+   * @param mixed $plugin_definition
+   * @param \Drupal\styleguide\GeneratorInterface $styleguide_generator
+   * @param \Drupal\Core\Theme\ThemeManagerInterface $theme_manager
+   * @param \Drupal\Core\Form\FormBuilder $form_builder
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   * @param \Drupal\search\SearchPluginManager $search_manager
+   *
+   * @internal param \Drupal\Core\Breadcrumb\ChainBreadcrumbBuilderInterface $breadcrumb
+   * @internal param \Drupal\styleguide\GeneratorInterface $generator
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, GeneratorInterface $styleguide_generator, ThemeManagerInterface $theme_manager, ModuleHandlerInterface $module_handler, FormBuilder $form_builder, AccountInterface $current_user, EntityManagerInterface $entity_manager, RendererInterface $renderer, SearchPluginManager $search_manager = NULL) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->generator = $styleguide_generator;
+    $this->themeManager = $theme_manager;
+    $this->moduleHandler = $module_handler;
+    $this->formBuilder = $form_builder;
+    $this->currentUser = $current_user;
+    $this->entityManager = $entity_manager;
```

Comment:
entity manager is deprecated, please use here entity type manager.
Details https://www.drupal.org/node/2549139